internal_event_handlers.go

```diff
@@ -335,6 +335,40 @@ func (wc *workflowEnvironmentImpl) RequestCancelTimer(timerID string) {
 	wc.logger.Debug("RequestCancelTimer", zap.String(tagTimerID, timerID))
 }
 
+func validateVersion(changeID string, version, minSupported, maxSupported Version) {
```

Comment:
Use error when invalid version is expected in legit case, and call site can handle that error accordingly. In this case, invalid version is not expected, and it is not recoverable. So I think panic is appropriate here.